### PR TITLE
Mes 4991/home tests

### PR DESCRIPTION
**Pull request by matthew2564**
Added fault providers for all Home test categories move getVehicleCheckFaultCount to common place to remove code duplication and added test

### CODE REVIEW

File: src/functions/sendCandidateResults/application/service/__tests__/fault-provider.spec.ts [hunk 1]
```diff
@@ -356,4 +357,77 @@ describe('fault-provider', () => {
...
+          },
+          {
+            outcome: 'DF',
+          },
+        ],
+      };
+      const count: number = getVehicleCheckFaultCount(data, CompetencyOutcome.DF);
+      expect(count).toEqual(4);
+    });
+  });
```
Comment (rossfauldsbjss) [inline src/functions/sendCandidateResults/application/service/__tests__/fault-provider.spec.ts:432]: Confused by the tests - for home tests there is only 1 show me 1 tell me, and even if you have 2 wrong results it only counts as 1  df.

File: src/functions/sendCandidateResults/application/service/categories/K/fault-provider-cat-k.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    throw new Error('No Test Data');
+  }
+
+  if (testData.dangerousFaults) {
+    convertBooleanFaultObjectToArray(testData.dangerousFaults)
+      .forEach(fault => dangerousFaults.push(fault));
+  }
+
+  return dangerousFaults;
+};
```
Comment (rossfauldsbjss) [inline src/functions/sendCandidateResults/application/service/categories/K/fault-provider-cat-k.ts]: Don't like the idea of a fault provider for each home test category... pretty sure a common home-test provider could be created. Avoids duplication

**Review by rossfauldsbjss — CHANGES_REQUESTED** [review]
Some stuff I think could be made common - e.g a HomeTest version that all the home test categories call. Most of them are identical, only minor difference for K but I don't think it affects the counts.

File: src/functions/sendCandidateResults/application/service/__tests__/fault-provider.spec.ts (hunk 1 above)
Comment (matthew2564) [reply src/functions/sendCandidateResults/application/service/__tests__/fault-provider.spec.ts:432]: This was the duplicated function I mentioned, its just doing a count of the number of specified outcomes in the show me/tell me questions array.

File: src/functions/sendCandidateResults/application/service/categories/K/fault-provider-cat-k.ts (hunk 2 above)
Comment (matthew2564) [reply src/functions/sendCandidateResults/application/service/categories/K/fault-provider-cat-k.ts]: Done

Comment (matthew2564) [reply src/functions/sendCandidateResults/application/service/categories/K/fault-provider-cat-k.ts]: done

**Review by rossfauldsbjss — APPROVED** [review]
looks good

File: src/functions/sendCandidateResults/application/service/categories/Home/__tests__/fault-provider-cat-home.spec.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+      const result: Fault [] = getSeriousFaultsCatHome(data);
+      expect(result.length).toBe(2);
+      expect(result).toContain({ name: Competencies.ancillaryControls, count: 1 });
+      expect(result).toContain({ name: Competencies.reverseLeftControl, count: 1 });
+    });
+  });
+
+  describe('getDrivingFaultsCatHome', () => {
+    it('should give us a list of faults from different sections of the test data', () => {
```
Comment (ammarhaiderbjss) [inline src/functions/sendCandidateResults/application/service/categories/Home/__tests__/fault-provider-cat-home.spec.ts]: Can we be a bit more specific with these it statements?

**Review by ammarhaiderbjss — APPROVED** [review]
Nice job

**Pull request merged**